sql/hive/src/test/scala/org/apache/spark/sql/hive/orc/HiveOrcQuerySuite.scala

```diff
@@ -415,4 +415,23 @@ class HiveOrcQuerySuite extends OrcQueryTest with TestHiveSingleton {
       }
     }
   }
+
+  test("SPARK-49094: ignoreCorruptFiles works for hive orc") {
+    withSQLConf(SQLConf.ORC_SCHEMA_MERGING_ENABLED.key -> "true") {
+      withTempDir { dir =>
+        val basePath = dir.getCanonicalPath
+        spark.range(0, 1).toDF("a").write.orc(new Path(basePath, "foo=1").toString)
+        spark.range(0, 1).toDF("b").write.json(new Path(basePath, "foo=2").toString)
+
+        withSQLConf(
+          SQLConf.IGNORE_CORRUPT_FILES.key -> "false",
+          SQLConf.ORC_IMPLEMENTATION.key -> "hive") {
+          checkAnswer(spark.read
+            .option("mergeSchema", value = false)
```

Comment:
To be complete, could you add another test case, `SQLConf.ORC_SCHEMA_MERGING_ENABLED.key -> "false"` without this line?

Reply:
```scala
         Seq(true, false).foreach { mergeSchema =>
            checkAnswer(spark.read
              .option("mergeSchema", value = mergeSchema)
              .option("ignoreCorruptFiles", value = true)
              .orc(basePath), Row(0L, 1))
          }
```
Using mergeSchema w/ true & false can also achieve your request. I will address this in this way.

Reply:
Unfortunately, no, it's different because the previous code doesn't care of `OrcOption`. I expect the exact request test case, @yaooqinn .

Reply:
> the previous code doesn't care of OrcOption

mergeSchema is actually from OrcOption for both if and else code branch

https://github.com/apache/spark/pull/47583/files#diff-9954c15e0525a72c38dd8be7eaff4ba6c2a071b216ecea2524419d49ab4f92fcR68-R69

Reply:
Oh, I realized the source of our disagreement. I overlooked the outermost withSQLConf wrapper, which is redundant.

Reply:
Does everything seem to be okay? The test now tests both mergeSchema on and off.